readthedocs/doc_builder/environments.py

```diff
@@ -452,7 +453,7 @@ def run_command_class(
             kwargs['bin_path'] = env_path
         if 'environment' in kwargs:
             raise BuildEnvironmentError('environment can\'t be passed in via commands.')
-        kwargs['environment'] = self.environment
+        kwargs['environment'] = self.environment.copy()
```

Comment:
Are we always calling `copy` so we don't accidentally modify it in memory?

Reply:
Yes